ios/brave-ios/scripts/scheme_preaction.py

```diff
@@ -0,0 +1,181 @@
+    target_arch = 'arm64' if platform.processor(
+    ) == 'arm' or options.platform_name == 'iphoneos' else 'x64'
+    target_environment = 'simulator' if (options.platform_name
+                                         == 'iphonesimulator') else None
+
+    UpdateSymlink(config, target_arch, target_environment)
+    BuildCore(config, target_arch, target_environment)
+    GenerateXCFrameworks(config, target_arch, target_environment)
+    CleanupChromiumAssets(output_dir)
+    FixMaterialComponentsVersionString(output_dir)
+    GenerateXcodeConfig(output_dir)
+    CallNpm(['npm', 'run', 'ios_pack_js'])
+
+
+def BuildOutputDirectory(config, platform_name):
+    directory_name = 'ios_%s' % config
+    if platform.processor() == 'arm' or platform_name == 'iphoneos':
+        directory_name += '_arm64'
+    if platform_name == 'iphonesimulator':
+        directory_name += '_simulator'
+    return os.path.join(os.path.join(src_dir, 'out'), directory_name)
+
+
+def UpdateSymlink(config, target_arch, target_environment):
+    """Updates the 'current_link' symlink"""
+    cmd_args = [
+        'npm', 'run', 'update_symlink', '--', config, '--symlink_dir',
+        os.path.join(src_dir, 'out/current_link'), '--target_os', 'ios',
+        '--target_arch', target_arch
+    ]
+    if target_environment != None:
+        cmd_args += ['--target_environment', target_environment]
+    CallNpm(cmd_args)
+
+
+def BuildCore(config, target_arch, target_environment):
+    """Generates and builds the BraveCore.framework"""
+    cmd_args = [
+        'npm',
+        'run',
+        'build',
+        '--',
+        config,
+        '--target_os',
+        'ios',
+        '--target_arch',
+        target_arch,
+    ]
+    if target_environment != None:
+        cmd_args += ['--target_environment', target_environment]
+    CallNpm(cmd_args)
+
+
+def _FrameworksForXCFramework(xcframework_dir):
+    """Returns a list of framework directories inside of a xcframework"""
+    framework_dirs = []
+    for root, dirs, _ in os.walk(xcframework_dir):
+        for folder in dirs:
+            if folder.endswith('.framework'):
+                framework_dirs += [
+                    os.path.join(xcframework_dir, os.path.join(root, folder))
+                ]
+    return framework_dirs
+
+
+def CleanupChromiumAssets(output_dir):
+    """Delete Chromium Assets from BraveCore.xcframework since they aren't
+    used."""
+    # TODO(@brave/ios): Get this removed in the brave-core builds if possible
+    xcframework_dir = os.path.join(output_dir, "BraveCore.xcframework")
+    for framework_dir in _FrameworksForXCFramework(xcframework_dir):
+        os.remove(os.path.join(framework_dir, "Assets.car"))
+
+
+def FixMaterialComponentsVersionString(output_dir):
+    """Adds a CFBundleShortVersionString to the outputted MaterialComponents
+    xcframework so that it's valid to upload"""
+    # TODO(@brave/ios): Fix this with a patch or chromium_src override somehow
```

Comment:
are we handling those todos now or later? if later we can add tickets for them

Reply:
Later, these todos are actually old and existed in the old bootstrap as well